framework/DataObjects/HistorySet.py

```diff
@@ -254,6 +275,24 @@ def _updateSpecializedOutputValue(self,name,value,options=None):
       @ In, value, ?, ?
       @ Out, None
     """
+
+    ## Check if we need to reduce the dataset
+    value = np.atleast_1d(value).flatten()
+    rows = None
+    if self._dataParameters is not None:
+      rows = self._dataParameters.get('outputRow', None)
+      ## Zero or one indexed?
+      # if rows  > 0:
+      #   rows  -= 1
+
+    if rows is None:
+      rows = range(len(value))
+    else:
+      if rows > len(value):
+        rows = range(len(value))
+        self.raiseAWarning("inputRow > len of history! Taking last row!")
+    value = np.atleast_1d(value[rows])
```

Comment:
you may need to create a method to get the value from given rows, because similar codes are used for both inputRow and outputRow selection.

Reply:
I think I will not do it here since we are moving to the new dataobject where this issue is addressed already.
Is it ok?

Reply:
It is fine to me.